PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-azure-sdk/src/main/java/com/microsoft/azure/toolkit/intellij/azure/sdk/buildtool/RuleConfigLoader.java

```diff
@@ -214,11 +229,46 @@ private Map<String, String> getMapOfDiscouragedIdentifiers(JsonReader reader, Ma
                 case "anti_pattern_message":
                     antiPatternMessage = reader.getString();
                     break;
+                default:
+                    reader.skipChildren();
             }
             if (identifiersToCheck != null && antiPatternMessage != null) {
                 discouragedIdentifiersToCheckMap.put(identifiersToCheck, antiPatternMessage);
             }
         }
         return discouragedIdentifiersToCheckMap;
     }
+
+    /**
+     * This method parses the map of anti-pattern messages from the JSON file
+     * This is used in rulechecks with 2 different antipattern messages depending on a certain condition.
+     *
+     * @param reader - the JsonReader object to read the JSON file
+     * @return Map of anti-pattern messages parsed from the JSON file
+     * @throws IOException - if there is an error reading the file
+     */
+    private Map<String, String> getAntiPatternMessageMap(JsonReader reader, Map<String, String> antiPatternMessageMap) throws IOException {
+
+        if (reader.nextToken() != JsonToken.START_OBJECT) {
+            throw new IOException("Expected start of object");
+        }
+
+        // Read the JSON file and parse the RuleConfig object
+        while (reader.nextToken() != JsonToken.END_OBJECT) {
+
+            // Get the field name
+            String fieldName = reader.getFieldName();
+
+            switch (fieldName) {
+                case "no_block", "with_subscribe" -> {
+                    String antiPatternMessage = reader.getString();
+                    if (antiPatternMessage != null) {
+                        antiPatternMessageMap.put(fieldName, antiPatternMessage);
+                    }
+                }
+                default -> reader.skipChildren();
```

Comment:
what is the Java baseline for the plugin? For example, in our Java SDK repro, all SDKs needs to be baselined to Java 8, is there a baseline requirement for the plugin? Here we are using enhanced switch block which is available only from Java 14.

Reply:
I'm working on this in Java 17, but I can use the normal switch block instead? 
`switch (fieldName) {
                case "no_block":
                case "with_subscribe":
                    antiPatternMessage = reader.getString();
                    break;`